tools/deno/bump-omicron.ts

```diff
@@ -56,29 +55,28 @@ function linkifyGitLog(line: string): string {
   return `* ${shaLink} ${rest}`
 }
 
-// script starts here
+async function makeOmicronWorktree() {
+  const tmpDir = await $`mktemp -d`.text()
+  await $`git worktree add ${tmpDir} origin/main`.cwd(OMICRON_DIR).quiet('stdout')
 
-const args = flags.parse(Deno.args, {
-  alias: { dryRun: ['d', 'dry-run'], h: 'help', m: 'message' },
-  boolean: ['dryRun', 'help'],
-  string: ['message'],
-})
-
-if (args.help) {
-  console.info(HELP)
-  Deno.exit()
+  return {
+    dir: tmpDir,
+    [Symbol.asyncDispose]: async function () {
+      console.log('Cleaning up worktree')
+      await $`git worktree remove ${tmpDir}`.cwd(OMICRON_DIR).quiet('stdout')
+    },
+  }
```

Comment:
this is really cool
https://www.totaltypescript.com/typescript-5-2-new-keyword-using